mlir/lib/Quantum/Transforms/LoopBoundaryOptimizationPatterns.cpp

```diff
@@ -0,0 +1,634 @@
+// Copyright 2025 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#define DEBUG_TYPE "loop-boundary"
+
+#include "llvm/ADT/StringSet.h"
+#include "llvm/Support/Debug.h"
+
+#include "mlir/Dialect/SCF/IR/SCF.h"
+#include "mlir/Dialect/Tensor/IR/Tensor.h"
+#include "mlir/IR/Operation.h"
+#include "mlir/IR/PatternMatch.h"
+
+#include "Quantum/IR/QuantumOps.h"
+#include "Quantum/Transforms/Patterns.h"
+
+using llvm::dbgs;
+using namespace llvm;
+using namespace mlir;
+using namespace catalyst;
+using namespace catalyst::quantum;
+
+namespace {
+
+// TODO: Add and test CRX, CRY, CRZ, ControlledPhaseShift, PhaseShift
+static const mlir::StringSet<> rotationsSet = {"RX", "RY", "RZ"};
+static const mlir::StringSet<> hamiltonianSet = {"Hadamard", "PauliX", "PauliY", "PauliZ",
```

Comment:
Minor, but I'm guessing you meant Hermitian (i.e. self-inverse) gates here?
```suggestion
static const mlir::StringSet<> hermitianSet = {"Hadamard", "PauliX", "PauliY", "PauliZ",
```

Reply:
You are right. It is supposed to be Hermitian.